dotnet/src/webdriver/DevTools/DevToolsSession.cs

```diff
@@ -437,116 +413,38 @@ private void OnTargetDetached(object sender, TargetDetachedEventArgs e)
             }
         }
 
-        private void TerminateSocketConnection()
+        private async Task InitializeSocketConnection()
         {
-            if (this.sessionSocket != null && this.sessionSocket.State == WebSocketState.Open)
-            {
-                var closeConnectionTokenSource = new CancellationTokenSource(this.closeConnectionWaitTimeSpan);
-                try
-                {
-                    // Since Chromium-based DevTools does not respond to the close
-                    // request with a correctly echoed WebSocket close packet, but
-                    // rather just terminates the socket connection, so we have to
-                    // catch the exception thrown when the socket is terminated
-                    // unexpectedly. Also, because we are using async, waiting for
-                    // the task to complete might throw a TaskCanceledException,
-                    // which we should also catch. Additiionally, there are times
-                    // when mulitple failure modes can be seen, which will throw an
-                    // AggregateException, consolidating several exceptions into one,
-                    // and this too must be caught. Finally, the call to CloseAsync
-                    // will hang even though the connection is already severed.
-                    // Wait for the task to complete for a short time (since we're
-                    // restricted to localhost, the default of 2 seconds should be
-                    // plenty; if not, change the initialization of the timout),
-                    // and if the task is still running, then we assume the connection
-                    // is properly closed.
-                    LogTrace("Sending socket close request");
-                    Task closeTask = Task.Run(async () => await this.sessionSocket.CloseOutputAsync(WebSocketCloseStatus.NormalClosure, string.Empty, closeConnectionTokenSource.Token));
-                    closeTask.Wait();
-                }
-                catch (WebSocketException)
-                {
-                }
-                catch (TaskCanceledException)
-                {
-                }
-                catch (AggregateException)
-                {
-                }
-            }
+            LogTrace("Creating WebSocket");
+            this.connection = new WebSocketConnection();
+            connection.DataReceived += OnConnectionDataReceived;
+            await connection.Start(this.websocketAddress);
+            LogTrace("WebSocket created; starting message listener");
         }
 
-        private void CancelReceiveTask()
+        private async void TerminateSocketConnection()
```

Comment:
Avoid `async void`, better is `async Task`.

Reply:
Of course, you’re right here. Thanks for catching the oversight.